packages/flow-core/src/components/f-text/f-text.scss

```diff
@@ -201,9 +204,73 @@ f-div[direction="row"] {
 }
 
 :host {
+	@include base();
 	mark {
 		background-color: transparent;
 		color: var(--color-highlight-default);
 		display: contents;
 	}
+	.f-text-editable {
+		display: flex;
+		flex-direction: row;
+		width: 100%;
+	}
+	.pseudo-edit-text {
+		display: none;
+		position: absolute;
+		top: 0;
+		right: 0;
+	}
+	.pseudo-submit-text {
+		position: absolute;
+		top: 5px;
+		right: 6px;
+	}
+	.non-editable-slot {
+		width: 95%;
+	}
+	.textarea {
+		display: block;
+		width: 100%;
+		overflow: hidden;
+		min-height: 15px;
+		// line-height: 20px;
+		border: 1px solid var(--color-border-default);
+		border-radius: 4px;
+		font-family: inherit;
+		font-size: inherit;
+		padding: 3px 6px;
+		resize: none;
+		.slot-text {
+			width: 95%;
+		}
+
+		@each $variant, $valmap in $variants {
+			&[variant="#{$variant}"] {
+				@each $size, $sizemap in $valmap {
+					&[size="#{$size}"] {
+						font-size: map-get($sizemap, "fontSize");
+						line-height: map-get($sizemap, "lineHeight");
+					}
+				}
+			}
+		}
+
+		@each $weight, $value in $weights {
+			&[weight="#{$weight}"] {
+				font-weight: $value;
+			}
+		}
+		&:focus {
+			outline: none;
+			border: 1px solid var(--color-primary-default);
+		}
+	}
+
+	.textarea[contenteditable]:empty::before {
+		@include base();
+		font-size: 12px;
+		content: "Enter Your Text here";
```

Comment:
can we use data attributes instead of static text?

Example https://tympanus.net/codrops/2013/07/05/using-custom-data-attributes-and-pseudo-elements/#:~:text=Let%E2%80%99s%20set%20the%20content%20for%20each%20pseudo%2Delement.%20We%E2%80%99ll%20get%20the%20value%20of%20the%20respective%20data%20attribute%20and%20add%20it%20to%20the%20content%20of%20the%20pseudo%2Delement%20by%20using%20attr()%3A